common/src/main/java/org/conscrypt/ConscryptFileDescriptorSocket.java

```diff
@@ -556,8 +556,16 @@ public int read(byte[] buf, int offset, int byteCount) throws IOException {
                     if (DBG_STATE) assertReadableOrWriteableState();
                 }
 
-                return NativeCrypto.SSL_read(sslNativePointer, Platform.getFileDescriptor(socket),
+                int ret = NativeCrypto.SSL_read(sslNativePointer, Platform.getFileDescriptor(socket),
                         ConscryptFileDescriptorSocket.this, buf, offset, byteCount, getSoTimeout());
+                if (ret == -1) {
+                    synchronized (stateLock) {
+                        if (state == STATE_CLOSED) {
+                            throw new SocketException("socket is closed");
```

Comment:
We can't acquire the lock in native code. Hopefully this class will be going away in short order.